### PR TITLE
A4A: Integrate Users REST endpoint to the team member list.

### PR DESCRIPTION
**Pull request by jkguidaven**
This PR wires the User's endpoint to the team members' page.

Screenshot 2024-08-27 at 7 13 09 PM

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/943

**Please ensure that this is only merged if D159377-code has been merged.**

## Proposed Changes

* Added a `useTeamMember` hook to handle both the fetch request for the users and the user invites endpoint. Consolidate the results into one member list, which is then outputted to the members table.
* Additionally, update the User invite hook to include additional fields (avatar icon and email).

## Why are these changes being made?

*

## Testing Instructions

* Apply the D159377-code to your sandbox and point `public-api.wordpress.com` to your sandbox.
* Use the A4A live link and go to the `/team` page.
* Confirm that the UI fetches data from the Users and Users Invite endpoints. 
* Add some users or create some invites.
* Confirm that it appears in the table.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/data/team/use-fetch-active-members.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+	return useQuery( {
+		queryKey: [ 'a4a-active-members', agencyId ],
+		queryFn: () =>
+			wpcom.req.get( {
+				apiNamespace: 'wpcom/v2',
+				path: `/agency/${ agencyId }/users`,
+			} ),
+		select: ( data ) => {
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (vitozev) [inline client/a8c-for-agencies/data/team/use-fetch-active-members.ts]: Are we going to update this type once the API is finalized?

File: client/a8c-for-agencies/sections/team/hooks/use-member-list.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { useCallback, useMemo } from 'react';
+import useFetchActiveMembers from 'calypso/a8c-for-agencies/data/team/use-fetch-active-members';
+import useFetchMemberInvites from 'calypso/a8c-for-agencies/data/team/use-fetch-member-invite';
```
Comment (vitozev) [inline client/a8c-for-agencies/sections/team/hooks/use-member-list.ts]: Can we update the file name to be plural?

File: client/a8c-for-agencies/data/team/use-fetch-active-members.ts (hunk 1 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/data/team/use-fetch-active-members.ts]: This has been addressed. I've added a type for this.

**Review by vitozev — APPROVED** [review]
Works as expected! Left a few comments that we can address later.

File: client/a8c-for-agencies/sections/team/hooks/use-member-list.ts (hunk 2 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/team/hooks/use-member-list.ts]: Good catch! This has been addressed.

**Pull request merged**